core-web/libs/sdk/react/src/lib/components/DotEditableText/DotEditableText.tsx

```diff
@@ -0,0 +1,135 @@
+import { Editor } from '@tinymce/tinymce-react';
+import { useEffect, useRef, useState } from 'react';
+
+import {
+    isInsideEditor as isInsideEditorFn,
+    postMessageToEditor,
+    CUSTOMER_ACTIONS,
+    DotCmsClient
+} from '@dotcms/client';
+
+import { DotEditableTextProps, TINYMCE_CONFIG } from './utils';
+
+export function DotEditableText({
+    mode = 'plain',
+    format = 'text',
+    contentlet,
+    fieldName = ''
+}: Readonly<DotEditableTextProps>) {
+    const editorRef = useRef<Editor['editor'] | null>(null);
+    const [isInsideEditor, setIsInsideEditor] = useState(false);
+    const [content, setContent] = useState(contentlet[fieldName] || '');
+
+    useEffect(() => {
+        const isInsideEditor = isInsideEditorFn();
+        setIsInsideEditor(isInsideEditor);
+
+        if (!isInsideEditor) {
+            return;
+        }
+
+        const content = contentlet[fieldName] || '';
+        editorRef.current?.setContent(content, { format });
+        setContent(content);
+    }, [contentlet, fieldName, format]);
+
+    useEffect(() => {
+        const onMessage = ({ data }: MessageEvent) => {
+            const { name, payload } = data;
+            if (name !== 'COPY_CONTENTLET_INLINE_EDITING_SUCCESS') {
+                return;
+            }
+
+            const { oldInode, inode } = payload;
+            const currentInode = contentlet.inode;
+            const shouldFocus = currentInode === oldInode || currentInode === inode;
+
+            if (shouldFocus) {
+                editorRef.current?.focus();
+            }
+        };
+
+        window.addEventListener('message', onMessage);
+
+        return () => {
+            window.removeEventListener('message', onMessage);
+        };
+    }, [contentlet.inode]);
+
+    const onMouseDown = (event: MouseEvent) => {
+        const { onNumberOfPages = 1 } = contentlet;
+
+        if (onNumberOfPages <= 1 || editorRef.current?.hasFocus()) {
+            return;
+        }
+
+        const { inode, languageId: language } = contentlet;
+
+        event.stopPropagation();
+        event.preventDefault();
+
+        try {
+            postMessageToEditor({
+                action: CUSTOMER_ACTIONS.COPY_CONTENTLET_INLINE_EDITING,
+                payload: {
+                    dataset: {
+                        inode,
+                        language,
+                        fieldName
+                    }
+                }
+            });
+        } catch (error) {
+            console.error('Failed to post message to editor:', error);
```

Comment:
Why this could fail? we need a more clear error message here...

Reply:
It shouldn't fail. I added the try/catch for a Copilot recommendation.

I improved the message.